### DISCUSSION TITLE
Language aware keybinding

### DISCUSSION THREAD

**Discussion by ze-gois** [zed-industries/zed#19259] [Feature Requests]
I am working with scripts - prototyping.

I am used to MATLAB and Jupyter keybindings.

To insert a cell we usually hit "ctrl+shift+-".

I am to reach the same effect in Zed.

Chatting with Claude Sonnet I was induced to insert to ~/.config/zed/keymap.json:

```JSON
 {
    "context": "Editor & languageId == 'python'",
    "bindings": {
      "ctrl-shift--": ["insertText", { "text": "\n## %% Cell\n" }]
    }
  },
  {
    "context": "Editor & languageId == 'rust'",
    "bindings": {
      "ctrl-shift--": ["insertText", { "text": "\n// %% Cell\n" }]
    }
  },
```

However it does not work.

Any advices?

**Comment by numist** [comment]
I think `&` should be `&&` in your example?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
